pkg/producers/s3/s3Producer.go

```diff
@@ -54,15 +54,15 @@ func (p *S3Producer) Produce(k []byte, v []byte, o any) {
 	if k == nil || len(k) == 0 {
```

Comment:
- there is no need to check `k==nil`, `len(k)` is `0` if `k==nil`
- you should check also if `k` is the string `"null"` with: `if len(k) == 0 || strings.ToLower(string(k)) == "null"...`


same applies to `gcs`